### PR TITLE
Fix MultipartFormDataParser construction errors

### PR DESCRIPTION
**Pull request by sallfris**
Fix Fatal error: Exception 'RuntimeException' with message 'Unable to read stream contents' on run MultipartFormDataParserTest::testParse()

| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  |

### CODE REVIEW

**Comment by sallfris** [comment]
Now all tests passed. Or it is necessary to create two pullrequests, first for MultipartFormDataParser.php and second for MultipartFormDataParserTest ?

**Comment by samdark** [comment]
What do you mean by all tests are passing? https://travis-ci.org/yiisoft/yii-web/jobs/481557824

File: src/MultipartFormDataParser.php [hunk 1]
```diff
@@ -192,7 +192,12 @@ public function parse($request)
                         } else {
                             fwrite($tmpResource, $value);
                             $fileConfig['tempFilename'] = $tmpFileName;
-                            $fileConfig['stream'] = new ResourceStream(['resource' => $tmpResource]); // save file resource, otherwise it will be deleted
+                            $fileConfig['stream'] = Yii::createObject([
```
Comment (hiqsol) [inline src/MultipartFormDataParser.php]: Why not simply: `new ResourceStream($tmpResource)` ?

Comment (sallfris) [reply src/MultipartFormDataParser.php]: Update

**Comment by samdark** [comment]
Seem to still fail: https://travis-ci.org/yiisoft/yii-web/jobs/481856344

**Comment by sallfris** [comment]
```
44) yii\web\tests\MultipartFormDataParserTest::testParse
ReflectionException: Class yii\http\ResourceStream does not have a constructor, so you cannot pass any constructor arguments

/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/AbstractContainer.php:346
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/AbstractContainer.php:158
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/AbstractContainer.php:134
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/AbstractContainer.php:155
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/AbstractContainer.php:134
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/di/src/Factory.php:48
/home/travis/build/yiisoft/yii-web/vendor/yiisoft/yii-core/src/helpers/BaseYii.php:69
/home/travis/build/yiisoft/yii-web/src/MultipartFormDataParser.php:198
/home/travis/build/yiisoft/yii-web/tests/MultipartFormDataParserTest.php:38
/home/travis/build/yiisoft/yii-web/vendor/phpunit/phpunit/phpunit:53
```
On the local server test passed

```
Testing started at 1:04 ...
D:\OSPanel\modules\php\PHP-7.2\php.exe D:/OSPanel/domains/yii/yii-web/vendor/phpunit/phpunit/phpunit --configuration D:\OSPanel\domains\yii\yii-web\phpunit.xml.dist yii\web\tests\MultipartFormDataParserTest D:\OSPanel\domains\yii\yii-web\tests\MultipartFormDataParserTest.php --teamcity
PHPUnit 7.5.2 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.10
Configuration: D:\OSPanel\domains\yii\yii-web\phpunit.xml.dist

Time: 174 ms, Memory: 4.00MB

OK (6 tests, 21 assertions)

Process finished with exit code 0
```
And why `Class yii\http\ResourceStream does not have a constructor, so you cannot pass any constructor arguments` if  https://github.com/yiisoft/yii-core/blob/master/src/http/ResourceStream.php#L50  ?

**Comment by samdark** [comment]
Seems to be related to dependencies: https://github.com/yiisoft/yii-web/blob/master/.travis.yml#L86. I think it will get green if `composer.lock` is committed...

**Comment by sallfris** [comment]
I should committed composer.lock?

**Comment by TomaszKane** [comment]
Probably yes:
https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control
?

**Pull request merged**